src/formatDate.js

```diff
@@ -7,8 +7,47 @@
  *
  * @returns {string}
  */
+
+const DIVIDER_IDX = 3;
+const YEARS_PERIOD = 30;
+const YEARS_2000S = 20;
+const YEARS_1900S = 19;
+const YEAR_SHORT = 'YY';
+const YEAR_LONG = 'YYYY';
+
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const divider = fromFormat[DIVIDER_IDX];
+  const dateParts = date.split(divider);
+  const dateConfig = {};
+
+  for (let i = 0; i < 3; i++) {
+    const datePart = dateParts[i];
+    const formatPart = fromFormat[i];
+
+    if (formatPart === YEAR_LONG) {
+      dateConfig.YY = datePart.slice(2);
+    }
+
+    if (formatPart === YEAR_SHORT) {
+      dateConfig.YYYY = createLongYear(datePart);
+    }
+
+    dateConfig[formatPart] = datePart;
```

Comment:
This line is overwriting the 'YY' or 'YYYY' key in the dateConfig object with the original date part, which is incorrect. You should only assign to dateConfig[formatPart] if formatPart is not 'YY' or 'YYYY'.